### PR TITLE
refactor: handle validation exception correctly

### PR DESCRIPTION
**Pull request by XVincentX**
Get rid of the throw statement and handle it correctly with the helper provided by the Option monad.

### CODE REVIEW

File: packages/http/src/validator/validators/utils.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { IPrismDiagnostic } from '@stoplight/prism-core';
 import { DiagnosticSeverity, Segment } from '@stoplight/types';
 import { getSemigroup } from 'fp-ts/lib/NonEmptyArray';
 import { getValidation } from 'fp-ts/lib/Either';
-import { option } from 'fp-ts/lib/Option';
+import { option, tryCatch, Option } from 'fp-ts/lib/Option';
```
Comment (StefanDywersant) [inline packages/http/src/validator/validators/utils.ts]: I would try to follow our code-style here and
```suggestion
import * as Option from 'fp-ts/lib/Option';
```

Especially when there are functions from different buckets.

File: packages/http/src/validator/validators/__tests__/headers.spec.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { query as registry } from '../../deserializers';
 import { HttpHeadersValidator } from '../headers';
 import * as validateAgainstSchemaModule from '../utils';
 import { assertRight, assertLeft } from '@stoplight/prism-core/src/__tests__/utils';
+import { some } from 'fp-ts/lib/Option';
```
Comment (StefanDywersant) [inline packages/http/src/validator/validators/__tests__/headers.spec.ts]:
```suggestion
import * as Option from 'fp-ts/lib/Option';
```

Although `some` is pretty typical for FP I would stick to our convention.

File: packages/http/src/validator/validators/__tests__/path.spec.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { path as registry } from '../../deserializers';
 import { HttpPathValidator } from '../path';
 import * as validateAgainstSchemaModule from '../utils';
 import { assertLeft, assertRight } from '@stoplight/prism-core/src/__tests__/utils';
+import { some } from 'fp-ts/lib/Option';
```
Comment (StefanDywersant) [inline packages/http/src/validator/validators/__tests__/path.spec.ts]: Ditto

**Comment by philsturgeon** [comment]
Not sure I have enough context to comment, as I float further away from code and more into product land I need screenshots or error messages that were fixed or something less fp-ish to comment on. :)

**Pull request merged**